hoomd/hpmc/IntegratorHPMC.h

```diff
@@ -205,6 +205,52 @@ class PatchEnergy
             throw std::runtime_error("PatchEnergy (base class) does not support launchKernel");
             }
         #endif
+
+        //! Sets the attached status
+        virtual void setAttached(bool attached)
+            {
+            m_attached = attached;
+            }
+
+        //! Returns the attached status
+        virtual bool getAttached()
+            {
+            return m_attached;
+            }
+
+        //! Sets the log only flag
+        virtual void setLogOnly(bool log)
+            {
+            m_log_only = log;
+            }
+
+        //! Returns the log only flag
+        virtual bool getLogOnly()
+            {
+            return m_log_only;
+            }
+
+        //! Notify when object is detached
+        virtual void notifyDetach()
+            {
+            setAttached(false);
+            }
+
+        //! Update the OBB tree  for union of particles
+        virtual void buildOBBTree()
+            {
+            }
+
+        //! Return a flag to trigger the OBB update
+        virtual bool getBuildOBB()
+            {
+            return m_build_obb;
+            }
+
+    protected:
+        bool m_attached = true;   //! Track whether we have attached to the Simulation object
+        bool m_log_only = false;  //! If true, only use patch energy for logging
+        bool m_build_obb = false; //! Flag to update the OBB tree for union of particles
```

Comment:
I don't understand why a HPMC integrator needs these flags? What do they control?

`m_attached` does not seem to be used.

The logging system in v3 has been completely rewritten. Objects shouldn't need a notion of a "logging only" state. You don't have to solve in this PR as we haven't solved it elsewhere, but it will eventually be possible to attach force/energy objects to a Simulation in a way that they are not attached to the integrator and thus don't get evaluated when evolving the system forward in time.

Reply:
Indeed, I tried to implement some hacky attachment protocol but I quickly rescinded. These were leftovers from that.